Sources/Security/Signing+ResponseVerification.swift

```diff
@@ -81,9 +80,13 @@ extension HTTPResponse where Body == Data {
             forCaseInsensitiveHeaderField: HTTPClient.ResponseHeader.signature.rawValue,
             in: headers
         ) else {
-            Logger.warn(Strings.signing.signature_was_requested_but_not_provided(request))
-
-            return .failed
+            let signatureRequested = request.nonce != nil
+            if signatureRequested {
+                Logger.warn(Strings.signing.signature_was_requested_but_not_provided(request))
+                return .failed
+            } else {
+                return .notRequested
```

Comment:
This might not be true for endpoints that don't need a nonce I think? I'm not too clear on how the backend will know whether to sign those endpoints though...

Reply:
The endpoints that can have signatures are marked with a decorator, either a normal signature or static. Static ones don’t need a nonce to produce a signature.